src/main/scala/org/renci/cam/LookupService.scala

```diff
@@ -0,0 +1,420 @@
+package org.renci.cam
+
+import com.typesafe.scalalogging.LazyLogging
+import io.circe.generic.auto._
+import io.circe._
+import org.apache.jena.query.QuerySolution
+import org.http4s.{HttpRoutes, Uri}
+import org.phenoscape.sparql.SPARQLInterpolation._
+import org.renci.cam.Biolink.{biolinkData, BiolinkData}
+import org.renci.cam.HttpClient.HttpClient
+import org.renci.cam.Server.EndpointEnv
+import org.renci.cam.Server.LocalTapirJsonCirce.jsonBody
+import org.renci.cam.Util.IterableSPARQLOps
+import org.renci.cam.domain.{BiolinkClass, BiolinkPredicate, IRI}
+import sttp.tapir.Endpoint
+import sttp.tapir.generic.auto._
+import sttp.tapir.server.http4s.ztapir.ZHttp4sServerInterpreter
+import sttp.tapir.ztapir.{endpoint, query}
+import zio.blocking.{effectBlockingIO, Blocking}
+import zio.config.ZConfig
+import zio.{Has, RIO, URIO, ZIO}
+
+import scala.io.Source
+import scala.language.implicitConversions
+
+/** The LookupService can be used to look up concepts within CAM-KP-API without needing to use either high-level TRAPI queries or low-level
+  * SPARQL queries. It is intended to provide a middle path to interrogate the database and to identify cases where the TRAPI interface
+  * (QueryService) doesn't provide the correct results.
+  */
+object LookupService extends LazyLogging {
+
+  /* DATA STRUCTURES */
+
+  /** This case class represents a single triple from our database.
+    *
+    * @param s
+    *   Subject
+    * @param p
+    *   Predicate
+    * @param o
+    *   Object
+    * @param g
+    *   Graph
+    */
+  case class ResultTriple(
+    s: String,
+    p: String,
+    o: String,
+    g: String
+  )
+
+  /** A helper method for converting a QuerySolution with ?s, ?p, ?o, ?g into a ResultTriple. */
+  def fromQuerySolution(result: QuerySolution): ResultTriple = ResultTriple(
+    result.get("s").toString,
+    result.get("p").toString,
+    result.get("o").toString,
+    result.get("g").toString
```

Comment:
Are these meant to be just any kind of RDF node? Not necessarily IRI or literal or whatever?

Reply:
Yup! I've added a comment to explain this in cba4556.